ethereum_history_api/circuits/lib/src/verifiers/transaction/rlp.nr

```diff
@@ -0,0 +1,61 @@
+use crate::{verifiers::transaction::MAX_TX_RLP_LENGTH, transaction::{TxType, TxPartial}};
+use crate::rlp::decode::{RlpList, decode_list};
+use crate::misc::{types::{PaddedValue, ADDRESS_LENGTH, U128_LENGTH}, bytes::byte_value};
+use dep::u2b::u64_to_u8;
+use dep::proof::const::HASH_LENGTH;
+
+struct FieldLayout {
+    field_count: u64,
+    nonce_idx: u8,
+    gas_limit_idx: u8,
+    to_idx: u8,
+    signature_idx: u8,
+}
+
+global TX_TYPE_TO_FIELD_LAYOUT = [
+    FieldLayout {field_count: 9, nonce_idx: 0, gas_limit_idx: 2, to_idx: 3, signature_idx: 6},
+    FieldLayout {field_count: 11, nonce_idx: 1, gas_limit_idx: 3, to_idx: 4, signature_idx: 8},
+    FieldLayout {field_count: 12, nonce_idx: 1, gas_limit_idx: 4, to_idx: 5, signature_idx: 9},
+    FieldLayout {field_count: 14, nonce_idx: 1, gas_limit_idx: 4, to_idx: 5, signature_idx: 11},
+];
+
+global MAX_TX_FIELDS_COUNT = 14;
+
+pub(crate) fn assert_tx_rlp_equals<MAX_DATA_LEN>(
+    tx_rlp: [u8; MAX_TX_RLP_LENGTH],
+    tx_type: TxType,
+    tx: TxPartial<MAX_DATA_LEN>
+) {
+    let tx_rlp_list: RlpList<MAX_TX_FIELDS_COUNT> = decode_list(tx_rlp);
+
+    let FieldLayout {field_count, nonce_idx, gas_limit_idx, to_idx, signature_idx} = TX_TYPE_TO_FIELD_LAYOUT[tx_type];
+    assert(tx_rlp_list.num_fields == field_count, "Invalid number of fields in tx RLP");
+
+    let nonce = u64_to_u8(tx.nonce);
+    tx_rlp_list.fragments[nonce_idx].assert_string_eq("Nonce", tx_rlp, Into::into(byte_value(nonce)));
```

Comment:
We still need to improve modeling I think, we should be able to do something like this:
```rust
rlp.assert_include("Nounce", nonce_idx, value)
```

rlp and rlp_list is consisting of the same data, rlp_list only make sense in context of rlp. 
Feels like it should be the same object (i.e. rlp_list should include (reference?) to (raw) rlp)

For future PR's

Reply:
Agree. For future PRs